### PR TITLE
Survey submission filters

### PR DESCRIPTION
**Pull request by richardolsson**
This PR adds pagination and more filtering options for the survey submissions listing.

### CODE REVIEW

File: src/actions/surveySubmission.js [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
-export function retrieveSurveySubmissions(surveyId = null) {
+export function retrieveSurveySubmissions(surveyId = null, linked = null, page = 0, perPage = 20) {
     return ({ dispatch, getState, z }) => {
-        let orgId = getState().org.activeId;
+        const filters = [];
+        const orgId = getState().org.activeId;
         let promise;
 
+        if (linked !== null) {
+            linked = linked? ((filters.linked == 'linked')? 1 : 0) : null;
```
Comment (olleolleolle) [inline src/actions/surveySubmission.js]: Style note: the positioning of the `?` operator in the ternary conditional tripped me up. "Is this some Babel precompile feature?"

File: src/store/surveySubmissions.js [hunk 2]
```diff
@@ -8,50 +8,68 @@ import {
...
-                submissionList: createList(),
-            };
+        }
+    }
+    else if (action.type == types.RETRIEVE_SURVEY_SUBMISSIONS + '_FULFILLED') {
+        return Object.assign({}, state, {
+            submissionList: updateOrAddListItems(state.submissionList, action.payload.data.data, {
+                isPending: false,
+                error: null,
+                lastPage: Math.max(state.submissionList.lastPage, action.meta.page),
```
Comment (olleolleolle) [inline src/store/surveySubmissions.js]: This got me narrowing my eyes a little "What do we _know_ about these properties? How?" I don't have the answer, yet.

`Math.max` really does not like `undefined`:

```
Math.max(1,2,undefined)
NaN
```

`null` is OK, though:

```
Math.max(1,2,null)
2
```

File: src/actions/surveySubmission.js (hunk 1 above)
Comment (richardolsson) [reply src/actions/surveySubmission.js]: Yeah, multi-level ternaries are not a great style choice. I'll reconsider in the future. :)

File: src/store/surveySubmissions.js (hunk 2 above)
Comment (richardolsson) [reply src/store/surveySubmissions.js]: True! I think this entire framework for pagination needs to be revamped and this is something to consider when we do so.

**Pull request merged**